pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -356,41 +356,43 @@ func ValidateVirtualMachineSpec(field *k8sfield.Path, spec *v1.VirtualMachineSpe
 }
 
 func validateDataVolumeTemplate(field *k8sfield.Path, spec *v1.VirtualMachineSpec) (causes []metav1.StatusCause) {
-	if len(spec.DataVolumeTemplates) > 0 {
-
-		for idx, dataVolume := range spec.DataVolumeTemplates {
-			if dataVolume.Name == "" {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueRequired,
-					Message: fmt.Sprintf("'name' field must not be empty for DataVolumeTemplate entry %s.", field.Child("dataVolumeTemplate").Index(idx).String()),
-					Field:   field.Child("dataVolumeTemplate").Index(idx).Child("name").String(),
-				})
-			}
+	if len(spec.DataVolumeTemplates) == 0 {
+		return causes
+	}
 
-			dataVolumeRefFound := false
-			for _, volume := range spec.Template.Spec.Volumes {
-				// TODO: Assuming here that PVC name == DV name which might not be the case in the future
-				if volume.VolumeSource.PersistentVolumeClaim != nil && volume.VolumeSource.PersistentVolumeClaim.ClaimName == dataVolume.Name {
-					dataVolumeRefFound = true
-					break
-				} else if volume.VolumeSource.DataVolume != nil && volume.VolumeSource.DataVolume.Name == dataVolume.Name {
-					dataVolumeRefFound = true
-					break
-				}
-			}
+	for idx, dataVolume := range spec.DataVolumeTemplates {
+		if dataVolume.Name == "" {
+			causes = append(causes, metav1.StatusCause{
+				Type:    metav1.CauseTypeFieldValueRequired,
+				Message: fmt.Sprintf("'name' field must not be empty for DataVolumeTemplate entry %s.", field.Child("dataVolumeTemplate").Index(idx).String()),
+				Field:   field.Child("dataVolumeTemplate").Index(idx).Child("name").String(),
+			})
+		}
 
-			if !dataVolumeRefFound {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueRequired,
-					Message: fmt.Sprintf("DataVolumeTemplate entry %s must be referenced in the VMI template's 'volumes' list", field.Child("dataVolumeTemplate").Index(idx).String()),
-					Field:   field.Child("dataVolumeTemplate").Index(idx).String(),
-				})
-			}
+		dataVolumeRefFound := checkDataVolumeReference(field, idx, dataVolume, spec.Template.Spec.Volumes)
+		if !dataVolumeRefFound {
+			causes = append(causes, metav1.StatusCause{
+				Type:    metav1.CauseTypeFieldValueRequired,
+				Message: fmt.Sprintf("DataVolumeTemplate entry %s must be referenced in the VMI template's 'volume' list", field.Child("dataVolumeTemplate").Index(idx).String()),
+				Field:   field.Child("dataVolumeTemplate").Index(idx).String(),
+			})
 		}
 	}
+
 	return causes
 }
 
+func checkDataVolumeReference(field *k8sfield.Path, idx int, dataVolume v1.DataVolumeTemplateSpec, volumes []v1.Volume) bool {
+	for _, volume := range volumes {
```

Comment:
Can you keep the TODO comment here please?

Reply:
Yes definitely, can I try and solve it as well?

Reply:
What change do you propose? Not sure there is anything to change about this right now.

Reply:
The current logic assusmes that the PVC name equals the DV name right, and according to the TODO statement it might not be the same case in the future, what if we introduce a 'map' like `pvctoDVmap` to store the mapping between PVC names and DV names.
Then when we are supposed to check for the presence of DVs in the 'Volumes' list, we consult this map to match the PVC names with DV names.

Reply:
@0xFelix @alicefr How does that solution sound for the TODO comment?